modules/govuk_jenkins/templates/jobs/deploy_app.yaml.erb

```diff
@@ -20,7 +20,9 @@
             export DEPLOY_TO="<%= @environment -%>"
             export DEPLOY_TASK="$DEPLOY_TASK"
             export TAG="$TAG"
-            export HOSTFILTER="$TARGET_MACHINES"
+            if [ "${TARGET_MACHINES:-all}" != "all" ]; then
```

Comment:
`if the default of "all" != "all"`?

Reply:
I get its purpose, it just hurt my brain.

Reply:
The default from the Jenkins job is "all", but we also want to handle the case where it's empty. So if it's empty, we default to "all" before doing the comparison.

Reply:
Added a comment, cos I can now see this line is kinda confusing 😅 Thanks for pointing it out.

Reply:
Oh wait, putting a comment there isn't gonna work without some hairy YAML escaping. Maybe I'd better not :(